ctrlf.el

```diff
@@ -1036,56 +1060,75 @@ Wrap around if necessary."
 
 ;;;;; Main commands
 
-(defun ctrlf-forward (style &optional preserve initial-contents position)
+(defun ctrlf-forward (style &optional preserve initial-contents
+                            position fallback)
```

Comment:
I noticed that most CTRLF functions check if they are called from the minibuffer and not in a search session and call `isearch-forward` and similar functions if so. Since I am now saving the fallback function the `ctrlf-style-alist`, I figured having this logic in `ctrlf-forward` is cleaner.